crypto/src/main/java/net/consensys/cava/crypto/sodium/Signature.java

```diff
@@ -421,6 +436,15 @@ public KeyPair(Signature.PublicKey publicKey, Signature.SecretKey secretKey) {
       return secretKey;
     }
 
+    /**
+     * See Signature.SecretKey#toBoxKeyPair()
+     * 
+     * @return Box.KeyPair generated from Signature.KeyPair
+     */
+    public Box.KeyPair toBoxKeyPair() {
```

Comment:
Not sure we need both `Signature.SecretKey#toBoxKeyPair()` and `Signature.KeyPair#toBoxKeyPair()` - I'd make it _either_ `Box.KeyPair.fromSignatureSecretKey(Signature.SecretKey secretKey)` _or_ `Box.KeyPair.fromSignatureKeyPair(Signature.KeyPair keyPair)` with my preference being the latter.

Reply:
Agreed, kept only `Box.KeyPair.fromSignatureKeyPair(Signature.KeyPair keyPair)`